### PR TITLE
[Component] Rename Data

### PR DESCRIPTION
**Pull request by lamriaimen**
According to the guidelines
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by lamriaimen** [comment]
[ci-build][with-all-tests]

**Comment by alxbilger** [comment]
To do: update the deprecation dates

File: Sofa/Component/MechanicalLoad/src/sofa/component/mechanicalload/TrianglePressureForceField.h [hunk 1]
```diff
@@ -49,20 +49,39 @@ class TrianglePressureForceField : public core::behavior::ForceField<DataTypes>
...
-    core::objectmodel::lifecycle::DeprecatedData dmax{ this, "v24.06", "v24.12", "dmax", "Plan selection using normal, dmin, dmax has been removed. Triangles should be selected using an Engine.Select and passed using Data triangleList" };
+    core::objectmodel::lifecycle::DeprecatedData d_normal{ this, "v24.06", "v24.12", "normal", "Plan selection using normal, dmin, dmax has been removed. Triangles should be selected using an Engine.Select and passed using Data triangleList" };
+    core::objectmodel::lifecycle::DeprecatedData d_dmin{ this, "v24.06", "v24.12", "dmin", "Plan selection using normal, dmin, dmax has been removed. Triangles should be selected using an Engine.Select and passed using Data triangleList" };
+    core::objectmodel::lifecycle::DeprecatedData d_dmax{ this, "v24.06", "v24.12", "dmax", "Plan selection using normal, dmin, dmax has been removed. Triangles should be selected using an Engine.Select and passed using Data triangleList" };
```
Comment (alxbilger) [inline Sofa/Component/MechanicalLoad/src/sofa/component/mechanicalload/TrianglePressureForceField.h]: I am not sure you should rename the `DeprecatedData` type

File: Sofa/Component/MechanicalLoad/src/sofa/component/mechanicalload/TrianglePressureForceField.inl [hunk 2]
```diff
@@ -142,31 +150,33 @@ void TrianglePressureForceField<DataTypes>::addDForce(const core::MechanicalPara
...
+    for (unsigned int i = 0; i < _triangleList.size(); ++i)
     {
         const auto& t = this->m_topology->getTriangle(_triangleList[i]);
 
         const auto& n0 = DataTypes::getCPos(x0[t[0]]);
         const auto& n1 = DataTypes::getCPos(x0[t[1]]);
         const auto& n2 = DataTypes::getCPos(x0[t[2]]);
 
+        my_subset[i].area = sofa::geometry::Triangle::area(n0, n1, n2);
+        my_subset[i].force= d_pressure.getValue() * my_subset[i].area;
```
Comment (alxbilger) [inline Sofa/Component/MechanicalLoad/src/sofa/component/mechanicalload/TrianglePressureForceField.inl]: This should not be here!

**Review by alxbilger — CHANGES_REQUESTED** [review]

File: Sofa/Component/MechanicalLoad/src/sofa/component/mechanicalload/TrianglePressureForceField.inl [hunk 3]
```diff
@@ -175,72 +185,70 @@ void TrianglePressureForceField<DataTypes>::initTriangleInformation()
...
 
+        d_trianglePressureMap.endEdit();
+    }
 
-template<class DataTypes>
-void TrianglePressureForceField<DataTypes>::draw(const core::visual::VisualParams* vparams)
-{
-    if (!p_showForces.getValue())
-        return;
+    template<class DataTypes>
```
Comment (alxbilger) [inline Sofa/Component/MechanicalLoad/src/sofa/component/mechanicalload/TrianglePressureForceField.inl]: Be careful with the indentations

**Comment by alxbilger** [comment]
Thanks for vigilance @hugtalbot.

@lamriaimen as you can see in the following report, most of the deprecation warnings are in targets that you probably did not enable. I suggest that you activate in CMake "internal" plugins (the ones provided with the sources in `applications/plugins`), and I take care of the external plugins (such as ModelOrderReduction).

https://ci.inria.fr/sofa-ci-dev/job/sofa-framework/job/PR-4771/CI_CONFIG=fedora_clang,CI_PLUGINS=options,CI_TYPE=release/13/FullBuildWarnings-Clang/category.2119183113/

File: Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/TetrahedronFEMForceField.inl [hunk 4]
```diff
@@ -43,60 +43,60 @@ TetrahedronFEMForceField<DataTypes>::TetrahedronFEMForceField()
...
+    , d_updateStiffnessMatrix(initData(&d_updateStiffnessMatrix, false, "updateStiffnessMatrix", ""))
+    , d_assembling(initData(&d_assembling, false, "computeGlobalMatrix", ""))
+    , d_plasticMaxThreshold(initData(&d_plasticMaxThreshold, (Real)0.f, "plasticMaxThreshold", "Plastic Max Threshold (2-norm of the strain)"))
+    , d_plasticYieldThreshold(initData(&d_plasticYieldThreshold, (Real)0.0001f, "plasticYieldThreshold", "Plastic Yield Threshold (2-norm of the strain)"))
+    , d_plasticCreep(initData(&d_plasticCreep, (Real)0.9f, "plasticCreep", "Plastic Creep Factor * dt [0,1]. Warning this factor depends on dt."))
+    , d_gatherPt(initData(&d_gatherPt, "gatherPt", "number of dof accumulated per threads during the gather operation (Only use in GPU version)"))
+    , d_gatherBsize(initData(&d_gatherBsize, "gatherBsize", "number of dof accumulated per threads during the gather operation (Only use in GPU version)"))
+    , d_drawHeterogeneousTetra(initData(&d_drawHeterogeneousTetra, false, "d_drawHeterogeneousTetra", "Draw Heterogeneous Tetra in different color"))
```
Comment (bakpaul) [inline Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/TetrahedronFEMForceField.inl]:
```suggestion
    , d_drawHeterogeneousTetra(initData(&d_drawHeterogeneousTetra, false, "drawHeterogeneousTetra", "Draw Heterogeneous Tetra in different color"))
```

File: Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/QuadBendingSprings.inl [hunk 5]
```diff
@@ -33,9 +33,10 @@ namespace sofa::component::solidmechanics::spring
 template<class DataTypes>
 QuadBendingSprings<DataTypes>::QuadBendingSprings()
     : StiffSpringForceField<DataTypes>()
-    , localRange( initData(&localRange, type::Vec<2,int>(-1,-1), "localRange", "optional range of local DOF indices. Any computation involving only indices outside of this range are discarded (useful for parallelization using mesh partitionning)" ) )
+    , d_localRange(initData(&d_localRange, type::Vec<2,int>(-1, -1), "d_localRange", "optional range of local DOF indices. Any computation involving only indices outside of this range are discarded (useful for parallelization using mesh partitionning)" ) )
```
Comment (bakpaul) [inline Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/QuadBendingSprings.inl]:
```suggestion
    , d_localRange(initData(&d_localRange, type::Vec<2,int>(-1, -1), "localRange", "optional range of local DOF indices. Any computation involving only indices outside of this range are discarded (useful for parallelization using mesh partitionning)" ) )
```

File: Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/TriangularBendingSprings.inl [hunk 6]
```diff
@@ -289,11 +289,11 @@ TriangularBendingSprings<DataTypes>::TriangularBendingSprings()
     , d_kd(initData(&d_kd, Real(1.0),"damping","uniform damping for the all springs"))
     , d_showSprings(initData(&d_showSprings, true, "showSprings", "option to draw springs"))
     , l_topology(initLink("topology", "link to the topology container"))
-    , edgeInfo(initData(&edgeInfo, "edgeInfo", "Internal edge data"))
+    , d_edgeInfo(initData(&d_edgeInfo, "d_edgeInfo", "Internal edge data"))
```
Comment (bakpaul) [inline Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/TriangularBendingSprings.inl]:
```suggestion
    , d_edgeInfo(initData(&d_edgeInfo, "edgeInfo", "Internal edge data"))
```

File: Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/TriangularBiquadraticSpringsForceField.inl [hunk 7]
```diff
@@ -126,21 +126,30 @@ void TriangularBiquadraticSpringsForceField<DataTypes>::applyEdgeCreation(Index
 }
 
 template <class DataTypes> TriangularBiquadraticSpringsForceField<DataTypes>::TriangularBiquadraticSpringsForceField()
-    : triangleInfo(initData(&triangleInfo, "triangleInfo", "Internal triangle data"))
-    , edgeInfo(initData(&edgeInfo, "edgeInfo", "Internal edge data"))
-    , _initialPoints(initData(&_initialPoints,"initialPoints", "Initial Position"))
+    : d_triangleInfo(initData(&d_triangleInfo, "d_triangleInfo", "Internal triangle data"))
```
Comment (bakpaul) [inline Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/TriangularBiquadraticSpringsForceField.inl]:
```suggestion
    : d_triangleInfo(initData(&d_triangleInfo, "triangleInfo", "Internal triangle data"))
```

File: Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/TriangularBiquadraticSpringsForceField.inl [hunk 8]
```diff
@@ -126,21 +126,30 @@ void TriangularBiquadraticSpringsForceField<DataTypes>::applyEdgeCreation(Index
...
+    , d_edgeInfo(initData(&d_edgeInfo, "d_edgeInfo", "Internal edge data"))
```
Comment (bakpaul) [inline Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/TriangularBiquadraticSpringsForceField.inl]:
```suggestion
    , d_edgeInfo(initData(&d_edgeInfo, "edgeInfo", "Internal edge data"))
```

File: Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/TriangularQuadraticSpringsForceField.inl [hunk 9]
```diff
@@ -108,19 +108,27 @@ void TriangularQuadraticSpringsForceField<DataTypes>::applyTriangleDestruction(I
...
+    , d_poissonRatio(initData(&d_poissonRatio, (Real)0.3, "poissonRatio", "Poisson ratio in Hooke's law"))
+    , d_youngModulus(initData(&d_youngModulus, (Real)1000., "youngModulus", "Young modulus in Hooke's law"))
+    , d_dampingRatio(initData(&d_dampingRatio, (Real)0., "dampingRatio", "Ratio damping/stiffness"))
+    , d_useAngularSprings(initData(&d_useAngularSprings, true, "useAngularSprings", "If Angular Springs should be used or not"))
     , lambda(0)
     , mu(0)
     , l_topology(initLink("topology", "link to the topology container"))
-    , triangleInfo(initData(&triangleInfo, "triangleInfo", "Internal triangle data"))
-    , edgeInfo(initData(&edgeInfo, "edgeInfo", "Internal edge data"))
+    , d_triangleInfo(initData(&d_triangleInfo, "d_triangleInfo", "Internal triangle data"))
```
Comment (bakpaul) [inline Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/TriangularQuadraticSpringsForceField.inl]:
```suggestion
    , d_triangleInfo(initData(&d_triangleInfo, "triangleInfo", "Internal triangle data"))
```

File: Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/TriangularQuadraticSpringsForceField.inl [hunk 10]
```diff
@@ -108,19 +108,27 @@ void TriangularQuadraticSpringsForceField<DataTypes>::applyTriangleDestruction(I
...
+    , d_edgeInfo(initData(&d_edgeInfo, "d_edgeInfo", "Internal edge data"))
```
Comment (bakpaul) [inline Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/TriangularQuadraticSpringsForceField.inl]:
```suggestion
    , d_edgeInfo(initData(&d_edgeInfo, "edgeInfo", "Internal edge data"))
```

File: Sofa/Component/SolidMechanics/TensorMass/src/sofa/component/solidmechanics/tensormass/TetrahedralTensorMassForceField.inl [hunk 11]
```diff
@@ -233,7 +233,7 @@ TetrahedralTensorMassForceField<DataTypes>::TetrahedralTensorMassForceField()
     , lambda(0)
     , mu(0)
     , l_topology(initLink("topology", "link to the topology container"))
-    , edgeInfo(initData(&edgeInfo, "edgeInfo", "Internal edge data"))
+    , edgeInfo(initData(&edgeInfo, "d_edgeInfo", "Internal edge data"))
```
Comment (bakpaul) [inline Sofa/Component/SolidMechanics/TensorMass/src/sofa/component/solidmechanics/tensormass/TetrahedralTensorMassForceField.inl]:
```suggestion
    , edgeInfo(initData(&edgeInfo, "edgeInfo", "Internal edge data"))
```

File: Sofa/Component/SolidMechanics/TensorMass/src/sofa/component/solidmechanics/tensormass/TriangularTensorMassForceField.inl [hunk 12]
```diff
@@ -223,7 +223,7 @@ void TriangularTensorMassForceField<DataTypes>::applyTriangleDestruction(const s
 }
 
 template <class DataTypes> TriangularTensorMassForceField<DataTypes>::TriangularTensorMassForceField()
-    : edgeInfo(initData(&edgeInfo, "edgeInfo", "Internal edge data"))
+    : edgeInfo(initData(&edgeInfo, "d_edgeInfo", "Internal edge data"))
```
Comment (bakpaul) [inline Sofa/Component/SolidMechanics/TensorMass/src/sofa/component/solidmechanics/tensormass/TriangularTensorMassForceField.inl]:
```suggestion
    : edgeInfo(initData(&edgeInfo, "edgeInfo", "Internal edge data"))
```

**Review by bakpaul — CHANGES_REQUESTED** [review]

**Comment by bakpaul** [comment]
I tried to look everywhere and spotted the same mistake as in the other PRs : you sometimes changed the factory name. It might still remain errors. Better check it twice @lamriaimen 

By the way you seem to have messed up a bit with the edge info data in the Tensor forcefiles, you have modified the comments but not the actual data, please take a look there. 

Except for those chages, the PR seems ok to me.

File: Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/FastTetrahedralCorotationalForceField.inl [hunk 13]
```diff
@@ -135,26 +135,39 @@ void FastTetrahedralCorotationalForceField<DataTypes>::createTetrahedronRestInfo
 
 template <class DataTypes> 
 FastTetrahedralCorotationalForceField<DataTypes>::FastTetrahedralCorotationalForceField()
-    : pointInfo(initData(&pointInfo, "pointInfo", "Internal point data"))
-    , edgeInfo(initData(&edgeInfo, "edgeInfo", "Internal edge data"))
-    , tetrahedronInfo(initData(&tetrahedronInfo, "tetrahedronInfo", "Internal tetrahedron data"))
+    : d_pointInfo(initData(&d_pointInfo, "pointInfo", "Internal point data"))
+    , d_edgeInfo(initData(&d_edgeInfo, "d_edgeInfo", "Internal edge data"))
```
Comment (bakpaul) [inline Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/FastTetrahedralCorotationalForceField.inl]:
```suggestion
    , d_edgeInfo(initData(&d_edgeInfo, "edgeInfo", "Internal edge data"))
```

File: Sofa/Component/SolidMechanics/TensorMass/src/sofa/component/solidmechanics/tensormass/TriangularTensorMassForceField.inl [hunk 14]
```diff
@@ -223,17 +223,19 @@ void TriangularTensorMassForceField<DataTypes>::applyTriangleDestruction(const s
 }
 
 template <class DataTypes> TriangularTensorMassForceField<DataTypes>::TriangularTensorMassForceField()
-    : edgeInfo(initData(&edgeInfo, "edgeInfo", "Internal edge data"))
+    : d_edgeInfo(initData(&d_edgeInfo, "d_edgeInfo", "Internal edge data"))
```
Comment (bakpaul) [inline Sofa/Component/SolidMechanics/TensorMass/src/sofa/component/solidmechanics/tensormass/TriangularTensorMassForceField.inl]:
```suggestion
    : d_edgeInfo(initData(&d_edgeInfo, "edgeInfo", "Internal edge data"))
```

**Review by bakpaul — COMMENTED** [review]
Still some wrong data names...

**Comment by lamriaimen** [comment]
Hey @bakpaul In the renaming data process I use the Refactor option in Clion (IDE), sometimes it does this that it even renames the data in the double quotes even though I specify that the renaming should only be for the code.sorry for that I rechecked all the code base even for the PRS that got merged I think now the code is clean

**Comment by bakpaul** [comment]
Ok I get that. Without such tool it would be difficult to make that kind of huge refacto anyway...
If you've added the two last modifications I asked then it is ready for me ! Good job !

**Review by bakpaul — APPROVED** [review]

**Comment by alxbilger** [comment]
@lamriaimen Can you take a look at `FastTetrahedralCorotationalForceField::tetrahedronInfo`. I think you used the wrong type compared to `d_tetrahedronInfo`

**Pull request merged**